packages/client/src/check-deployment-status.ts

```diff
@@ -46,16 +52,51 @@ export async function* checkDeploymentStatus(
   const finishedEvents = new Set();
   const startTime = Date.now();
 
+  // Deployment polling
   while (true) {
-    // Deployment polling
-    const deploymentData = await fetch(
-      `${apiDeployments}/${deployment.id || deployment.deploymentId}${
-        teamId ? `?teamId=${teamId}` : ''
-      }`,
-      token,
-      { apiUrl, userAgent, agent: clientOptions.agent }
-    );
-    const deploymentUpdate = await deploymentData.json();
+    let deploymentResponse: any;
+    let retriesLeft = RETRY_COUNT;
+    while (true) {
+      deploymentResponse = await fetch(
+        `${apiDeployments}/${deployment.id || deployment.deploymentId}${
+          teamId ? `?teamId=${teamId}` : ''
+        }`,
+        token,
+        { apiUrl, userAgent, agent: clientOptions.agent }
+      );
+
+      retriesLeft--;
+      if (retriesLeft == 0) {
+        break;
+      }
+
+      // HTTP 429: Too Many Requests
+      if (deploymentResponse.status === 429) {
+        let retryAfterMs =
+          parseInt(deploymentResponse.headers.get('Retry-After'), 10) * 1000;
+        if (Number.isNaN(retryAfterMs)) {
+          retryAfterMs = RETRY_DELAY_DEFAULT_MS;
+        }
+        retryAfterMs = Math.min(
+          RETRY_DELAY_MAX_MS,
+          Math.max(RETRY_DELAY_MIN_MS, retryAfterMs)
+        );
+        await sleep(retryAfterMs);
```

Comment:
```suggestion
       debug(`Rate limited by the server while querying deployment status, retrying after ${retryAfterMs}ms`)
        await sleep(retryAfterMs);
```

something like that?